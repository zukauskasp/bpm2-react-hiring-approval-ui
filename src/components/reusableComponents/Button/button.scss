.customButton {
    cursor: pointer;
    height: 48px;
    margin: 0 9px 9px 9px;
    display: inline-block;
    padding: 0 32px;
    overflow: hidden;
    font-size: 16px;
    max-width: 282px;
    min-width: 144px;
    box-shadow: none;
    font-style: normal;
    text-align: center;
    transition: 0.3s;
    font-family: 'Danske Text v2';
    font-weight: 700;
    line-height: 48px;
    border-radius: 24px;
    text-transform: capitalize;
}

.basicIcon {
    display: inline-block;
    height: 1em;
    overflow: visible;
    vertical-align: -0.2em;
    width: 1em;
}