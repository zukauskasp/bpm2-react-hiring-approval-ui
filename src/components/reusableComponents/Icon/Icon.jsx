import React from "react";
import PropTypes from "prop-types";

const Icon = ({ icon, color, ...otherProps }) => {

    return (
        <img alt="" src={process.env.PUBLIC_URL + '/assets/icons/' + icon + '/' + color + '.svg'} {...otherProps} />
    )
}

Icon.propTypes = {
    icon: PropTypes.oneOf(['add', 'alert', 'back', 'check',
        'collapse', 'duplicate', 'edit', 'expand', 'info', 'next',
        'remove', 'removeInCircle', 'search', 'trash', 'upload'
    ]),
    color: PropTypes.oneOf(['white', 'dark']),
    otherProps: PropTypes.any,
}

Icon.defaultProps = {
    color: "dark"
}


export default Icon;