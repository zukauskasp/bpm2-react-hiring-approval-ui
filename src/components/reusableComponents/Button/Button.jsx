import React, { useState, useEffect } from "react";
import './button.scss';
import { makeStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import Icon from "../Icon/Icon";

const Button = ({ icon, buttonType, background, text, onClickEvent }) => {

    const classes = useStyles();
    const [iconColor, setIconColor] = useState(null)


    useEffect(() => {
        if (icon) setIconColor(buttonType === "secondary" && background === "light" ? 'dark' : 'white');
    }, [icon, buttonType, background]);

    return (
        <div
            onClick={onClickEvent}
            className={
                `customButton
                 ${buttonType === 'secondary' ? classes.customSecondaryButton : ""}
                 ${buttonType === 'primary' ? classes.customPrimaryButton : ""}
                 ${background === 'dark' && buttonType === 'secondary' ? classes.customSecondaryButtonDarkBg : ""}
                 ${icon ? classes.buttonWithIcon : ""}
                 ${!text ? classes.roundButton : ""}
                 `
            }
        >
            {text}
            {icon ? <Icon color={iconColor} icon={icon} className={`basicIcon ${text ? classes.textIcon : ""} `} /> : null}
        </div>
    );

}

Button.propTypes = {
    icon: PropTypes.oneOf(['add', 'remove', 'search', 'next', 'expand']),
    buttonType: PropTypes.oneOf(['primary', 'secondary']),
    background: PropTypes.oneOf(['light', 'dark']),
    text: PropTypes.string,
    onClickEvent: PropTypes.func.isRequired,
}

Button.defaultProps = {
    buttonType: "primary",
    background: "light",
    text: ""
}

export default Button;

const useStyles = makeStyles({
    customPrimaryButton: {
        background: '#009FDA',
        color: '#fff',
        '&:hover': {
            background: '#0283b4',
        },
    },
    customSecondaryButton: {
        background: 'rgba(197, 200, 203, 0.1)',
        border: '0.5px solid rgba(197, 200, 203, 0.5)',
        color: '#003755',
        '&:hover': {
            background: 'rgba(197, 200, 203, 0.3)'
        },
    },
    customSecondaryButtonDarkBg: {
        background: 'rgba(197, 200, 203, 0)',
        color: '#fff',
        border: '0.5px solid rgba(255, 255, 255, 0.3)',
        '&:hover': {
            background: 'rgba(197, 200, 203, 0.2)'
        },
    },
    buttonWithIcon: {
        padding: '0 24px 0 32px',
    },
    textIcon: {
        marginLeft: '16px',

    },
    roundButton: {
        borderRadius: '50%',
        color: '#fff',
        width: '48px',
        minWidth: '0',
        padding: 0,
        height: '48px',
        background: '#003755',
        '&:hover': {
            background: 'rgba(0,55,85,.9)'
        }
    }
});