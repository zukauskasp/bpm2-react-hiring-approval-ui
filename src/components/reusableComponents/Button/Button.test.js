import React from 'react';
import { shallow, mount } from 'enzyme';
import Button from './Button';
import Icon from "../Icon/Icon";

describe('Button', () => {

    afterAll(() => {
        mount.cleanUp();
    });

    it('expect to match latest snapshot', () => {
        const wrapper = mount(<Button icon="add" buttonType="secondary" onClickEvent={() => { }} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it('expect to render button component', () => {
        expect(shallow(<Button onClickEvent={() => { }} />).length).toEqual(1);
    });

    it('expect to have text from props', () => {
        const wrapper = shallow(<Button text="Button text" onClickEvent={() => { }} />);
        expect(wrapper.contains('Button text')).toEqual(true);
    });

    it('expect to have custom icon element rendered', () => {
        const wrapper = shallow(<Button icon="add" buttonType="secondary" background='light' onClickEvent={() => { }} />);
        expect(wrapper.find(Icon)).toHaveLength(1);
    });

    it('expect icon name is equal to button props', () => {
        const wrapper = shallow(<Button icon="add" buttonType="secondary" background='light' onClickEvent={() => { }} />);
        expect(wrapper.find(Icon).prop('icon')).toEqual('add')
    });

    it('expect to don"t have image element rendered', () => {
        const wrapper = mount(<Button buttonType="secondary" onClickEvent={() => { }} />);
        expect(wrapper.find('.basicIcon').length).toEqual(0);
    });
});