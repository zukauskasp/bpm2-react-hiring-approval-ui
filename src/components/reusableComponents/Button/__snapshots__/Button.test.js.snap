// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button expect to match latest snapshot 1`] = `
"<Button icon=\\"add\\" buttonType=\\"secondary\\" onClickEvent={[Function: onClickEvent]} background=\\"light\\" text=\\"\\">
  <div onClick={[Function: onClickEvent]} className=\\"customButton\\\\n                 makeStyles-customSecondaryButton-2\\\\n                 \\\\n                 \\\\n                 makeStyles-buttonWithIcon-4\\\\n                 makeStyles-roundButton-6\\\\n                 \\">
    <Icon color=\\"dark\\" icon=\\"add\\" className=\\"basicIcon  \\">
      <img alt=\\"\\" src=\\"/assets/icons/add/dark.svg\\" className=\\"basicIcon  \\" />
    </Icon>
  </div>
</Button>"
`;
